src/libraries/System.Private.CoreLib/src/System/Single.cs

```diff
@@ -183,6 +183,7 @@ public int CompareTo(object? value)
             throw new ArgumentException(SR.Arg_MustBeSingle);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
@tannergooding, @pgovind, should we be doing the same for Half.CompareTo?

Reply:
I don't think its as importantas `Half.CompareTo` is quite a bit more expensive. Each operator is an actual call, not a primitive operation that compiles down to a CPU instruction